src/aapns/api.py

```diff
@@ -0,0 +1,134 @@
+import json
+import ssl
+from asyncio import get_event_loop, wait_for, Lock
+from typing import Optional
+
+from structlog import BoundLogger
+
+from . import errors, config, models, connection
+
+
+
+class APNS:
+    def __init__(self, client_cert_path, server, logger, ssl_context, *, auto_reconnect=False, timeout=None):
+        self.client_cert_path = client_cert_path
+        self.server = server
+        self.logger = logger
+        self.ssl_context = ssl_context
+        self.auto_reconnect = auto_reconnect
+        if timeout is None:
+            self.run_with_timeout = self.run_without_timeout
```

Comment:
🤔  I'm not a huge fan of the magical "run_with_timeout"/"run_without_timeout" so maybe your suggestion is better.